src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -124,81 +124,118 @@ export default class SourceFileLinter {
 				context: this.#context,
 				checker: this.#checker,
 			});
-		} else if (ts.isPropertyDeclaration(node) && node.name.getText() === "metadata") {
+		} else if (
+			ts.isPropertyDeclaration(node) && node.name.getText() === "metadata" &&
+			ts.isClassDeclaration(node.parent) && this.isUi5ClassDeclaration(node.parent, "sap/ui/base/ManagedObject")
+		) {
 			const visitMetadataNodes = (childNode: ts.Node) => {
 				if (ts.isPropertyAssignment(childNode)) { // Skip nodes out of interest
 					this.analyzeMetadataProperty(childNode.name.getText(), childNode);
 				}
 
 				ts.forEachChild(childNode, visitMetadataNodes);
 			};
-
-			if (this.isUi5controlMetadataNode(node)) {
-				ts.forEachChild(node, visitMetadataNodes);
-			}
+			ts.forEachChild(node, visitMetadataNodes);
+		} else if (ts.isClassDeclaration(node) && this.isUi5ClassDeclaration(node, "sap/ui/core/Control")) {
```

Comment:
would it make sense to make the isClassDeclaration check part of the isUi5ClassDeclaration check?